### ISSUE TITLE
Build 372 blocked thread server crash, reproducible on 1.15.2 and 1.16.3 using latest goPaint sphere brush

### ISSUE THREAD

**Issue by Combustible** [IntellectualSites/FastAsyncWorldEdit#646]
# Bug report for FastAsyncWorldEdit 1.16.3 / 1.15.2 (reproducible on both)

**FastAsyncWorldEdit Configuration Files:** 
https://athion.net/ISPaster/paste/view/9f53e1f9f5c946c788a4587928f70ef9

**FastAsyncWorldEdit Version Number:** 
Broken in version FastAsyncWorldEdit-372
This used to work correctly in FastAsyncWorldEdit-349

**Spigot/Paper Version Number:** 
Paper version git-Paper-"d802a50a9" (MC: 1.16.3)
Paper version git-Paper-"cebec202f" (MC: 1.15.2)
GoPaint version (v3.0.0-dev) - Downloaded from https://ci.athion.net/view/Everything/job/goPaint-1.14+/ build 3

**Description of the problem:**
Using GoPaint sphere brush crashes server with blocked thread. This happens with other operations too, but this is the easiest one to reproduce I have found so far. No recovery is possible because this is on the main thread - server will die without being able to save any changes. 

[23:11:44 ERROR]: Current Thread: Server thread
[23:11:44 ERROR]:       PID: 17 | Suspended: false | Native: false | State: BLOCKED
[23:11:44 ERROR]:       Stack:
[23:11:44 ERROR]:               com.sk89q.worldedit.LocalSession.remember(LocalSession.java:479)
[23:11:44 ERROR]:               com.sk89q.worldedit.LocalSession.remember(LocalSession.java:408)
[23:11:44 ERROR]:               net.arcaniax.gopaint.objects.other.BlockPlacer.placeBlocks(BlockPlacer.java:68)
[23:11:44 ERROR]:               net.arcaniax.gopaint.objects.brush.SphereBrush.paint(SphereBrush.java:73)
[23:11:44 ERROR]:               net.arcaniax.gopaint.listeners.InteractListener.onClick(InteractListener.java:94)

**How to replicate**:
Get feather
Left click to open GUI
Put block that's on the ground (i.e. a grass block) into the mask on the right
Put block you want to sphere with (i.e. dirt) into the block on the left
Enable brush
Sphere is the default mode

**Plugins being used on the server:**

Plugins (18): Arceon, BKCommonLib, BungeeTabListPlus, CommandAPI,  (WorldEdit)FastAsyncWorldEdit, goBrush, goPaint, LightCleaner, LuckPerms, MonumentaWarps*, PlaceholderAPI, PrometheusExporter*, ProtocolLib, ScriptedQuests, SpeedChanger*, Vault, VentureChat, VoxelSniper

**Checklist**:
- [X] I included all information required in the sections above
- [X] I made sure there are no duplicates of this report [(Use Search)](https://github.com/IntellectualSites/FastAsyncWorldEdit/issues?q=is%3Aissue)
- [X] I made sure I am using an up-to-date version of [FastAsyncWorldEdit for 1.16.2](https://ci.athion.net/job/FastAsyncWorldEdit-1.16/)
- [X] I made sure the bug/error is not caused by any other plugin

**Comment by Combustible** [comment]
After more testing, I'm sure that this issue was introduced in build 372. Build 371 works.

**Comment by dordsor21** [comment]
> After more testing, I'm sure that this issue was introduced in build 372. Build 371 works.

no

**Comment by dordsor21** [comment]
Referenced wrong commit. goPaint is doing weird stuff. Not sure it's a FAWE problem at all.

**Comment by dordsor21** [comment]
The full stacktrace for thread blocks is almost always required.

**Comment by dordsor21** [comment]
This might be fixed. If not, please give the full thread dump (and once the server has completely crashed, so we can see which other threads are waiting on the same monitor as main thread, and thus causing the crash)

**Comment by aurorasmiles** [comment]
No reply for over 2 weeks, I assume this has been fixed.

**Comment by Combustible** [comment]
Was fixed, thanks!
